CMakeLists.txt

```diff
@@ -57,16 +57,38 @@ set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 
+# Conan dependencies
+include(conan)
+set(DYNAMIC_LIB_TRUE_FALSE "False")
+if(${DYNAMIC_LIB})
+  set(DYNAMIC_LIB_TRUE_FALSE "True")
+endif()
+
+# Detect compiler, build type etc from cmake settings
+conan_cmake_autodetect(conan_settings)
+# cmake-format: off
+conan_cmake_run(
+  REQUIRES eigen/3.3.9
+           gtest/1.11.0
+           benchmark/1.5.3
+  OPTIONS gtest:shared=${DYNAMIC_LIB_TRUE_FALSE}
+          benchmark:shared=${DYNAMIC_LIB_TRUE_FALSE}
+  SETTINGS ${conan_settings}
+  IMPORTS include/eigen3,*->./eigen-src
```

Comment:
IMPORTS provides means of grabbing files from the Conan cache. In this case the eigen source, but could be compiled shared lib files etc.